Feature/day4 assignment:Complete Day 4 Assignment

PranaliM0 opened the pull request:
Below is attached screenshot of today's Day 4 assignment
Assignment - 
	A CLI application for handling data of as many countries as we can. Example of what data that we can have:
Population
GDP(Gross Domestic Production)
States
Army Strength
State of country(developing, developed)
Etc.
On the basis of all these data we will be building modules to conceive various outputs like:
Whether that country will get a loan from the IMF (International Monetary Fund) or WB (World bank).
Whether the country can have a seat in the UN (United Nations) Security council.
Whether the country can win the war or not

![Country1](https://github.com/user-attachments/assets/31fbd41a-2876-4f24-860d-6ea1871096cb)
![Country2](https://github.com/user-attachments/assets/f39ec24d-bb62-42da-bc05-c77b028de904)
![Country3](https://github.com/user-attachments/assets/7f3759fd-8ea9-4ae6-b5ec-f0dd05039a3c)
![Country4](https://github.com/user-attachments/assets/64af2df1-ee53-4209-806a-a70c9ceb046e)
![Country5](https://github.com/user-attachments/assets/8c499ff0-070f-4099-87c5-796d64d51cac)
![Country6](https://github.com/user-attachments/assets/e87c2e6e-4bde-4c5f-9438-7adcf97e6838)

shubhh139 commented on country.rb:
```diff
@@ -0,0 +1,141 @@
+    else
+      "#{@name} is not eligible for loan"
+    end
+  end
+end
+module UnSecurityCouncil
```
@PranaliM0 Please add an empty line between two functions or any two modules for better readability

PranaliM0 replied:
okay!

shubhh139 commented on country.rb:
```diff
@@ -0,0 +1,141 @@
+end
+module UnSecurityCouncil
+  def un_security_council_eligibility?
+    if @army_strength>500_000 && @gdp > 2_000_000
```
@PranaliM0 You can use constants for the numbers with which you're doing comparison

PranaliM0 replied:
Okay,got it!

shubhh139 commented on country.rb:
```diff
@@ -0,0 +1,141 @@
+def find_country(name)
+  $all_countries.find {|c| c.name.downcase == name.downcase}    
+end
+
```
@PranaliM0 Instead of adding modules, classes in same file, you can also use separate files for implementation

shubhh139 commented on country.rb:
```diff
@@ -0,0 +1,141 @@
+  end
+end
+
+menu
```
@PranaliM0 Instead of function you can use class for this

PranaliM0 commented:
Refactored the code by creating separate files for each module (LoanEligibility, UnSecurityCouncil, WarPrediction) to improve modularity and maintainability. Replaced magic numbers with constants for better clarity, and made the Country class more organized by including these modules for specific functionalities. Improved readability by adding empty lines between functions and modules, and updated the main menu for better user interaction with the refactored code.
